File: client/python/mesh/prsim/evaluator.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from abc import ABC, abstractmethod
from typing import Generic, Any, Dict, List

from mesh.kinds import Script, Paging, Page
from mesh.macro import mpi, spi, T, A


@spi("mesh")
class Evaluator(ABC, Generic[T, A]):
    """
    Evaluate engine.
    """

    @abstractmethod
    @mpi("mesh.eval.compile")
    def compile(self, script: Script) -> str:
        """
        Compile the named rule.
        :param script:
        :return:
        """

    @abstractmethod
    @mpi("mesh.eval.exec")
    def exec(self, code: str, args: Any, dft: str) -> str:
        """
        Exec the script with name.
        :param code:
        :param args:
        :param dft:
        :return:
        """

    @abstractmethod
    @mpi("mesh.eval.dump")
    def dump(self, feature: Dict[str, str]) -> List[Script]:
        """
        Dump the scripts.
        :param feature:
        :return:
        """

    @abstractmethod
    @mpi("mesh.eval.index")
    def index(self, index: Paging) -> Page[Script]:
        """
        Dump the scripts.
        :param index:
        :return:
        """

File: client/java/src/main/java/io/be/mesh/prsim/Evaluator.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package io.be.mesh.prsim;

import io.be.mesh.macro.MPI;
import io.be.mesh.macro.SPI;
import io.be.mesh.struct.Page;
import io.be.mesh.struct.Paging;
import io.be.mesh.struct.Script;

import java.util.List;
import java.util.Map;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public interface Evaluator {

    /**
     * Compile the named rule.
     */
    @MPI("mesh.eval.compile")
    String compile(Script script);

    /**
     * Exec the script with code.
     */
    @MPI("mesh.eval.exec")
    String exec(String code, Map<String, String> args, String dft);

    /**
     * Dump the scripts.
     */
    @MPI("mesh.eval.dump")
    List<Script> dump(Map<String, String> feature);

    /**
     * Index the scripts.
     */
    @MPI("mesh.eval.index")
    Page<Script> index(Paging index);
}
